docs/protocol.md

```diff
@@ -1,26 +1,83 @@
-# TChannel v2
+# Specification
 
-## Design goals
+## Introduction
+
+### Problem Statement
+
+TChannel was developed at Uber during a period of explosive growth in its
+[service oriented architecture][soa]. Services living in a large, distributed
+system like this usually encounter similar classes of problems, such as:
+
+ - Service discovery -- how do I discover *and consume* the services around me?
+
+ - Fault tolerance -- how can I best insulate myself from failures outside of
+   my control?
+
+ - [Dapper][dapper]-style tracing -- how can I identify and monitor bottlenecks
+   across the entire system?
+
+[soa]: https://en.wikipedia.org/wiki/Service-oriented_architecture "SOA"
+[dapper]: http://research.google.com/pubs/pub36356.html "Dapper"
+
+### Design Goals
+
+TChannel aims to solve these problems by providing a protocol for clients and
+servers, with an *intelligent routing mesh* (referred to as *Hyperbahn*)
+connecting the two.
+
+Consider those [SOA][soa] problems again:
+
+ - Service discovery: All producers and consumers register themselves with the
+   routing mesh. Consumers access producers only by their name; no need to know
+   about hosts or ports.
+
+ - Fault tolerance: The routing mesh tracks things like failure rates and
+   [SLA][sla] violations. It can intelligently detect unhealthy hosts and
+   remove them from the pool of available hosts.
+
+ - Request tracing: Built into the protocol as a first class citizen.
+
+[sla]: https://en.wikipedia.org/wiki/Service-level_agreement "SLA"
+
+By consolidating logic *between* producers and consumers, this also allows us
```

Comment:
Prefer shorter, active, and more direct form: rather than "By consolidating... this allows us to ...", try "Consolidating allows us to..."